### PR TITLE
br/lightning: migrate chekpoints tests to testify

### LINKED ISSUE #28235: migrate test-infra to testify for `br/pkg/lightning/checkpoints`

### PR DESCRIPTION
**Pull request by disksing**
Signed-off-by: disksing <i@disksing.com>

### What problem does this PR solve?

Issue Number: close #28235

Problem Summary: Migrate tests of lightning/checkpoints to using testify

### Check List

Tests 
- [x] Unit test

### Release note
```release-note
None
```

### CODE REVIEW

**Review by tisonkun — COMMENTED** [review]
Generally looks good. I tend to use `require` consistently as mentioned above.

File: br/pkg/lightning/checkpoints/checkpoints_file_test.go [hunk 1]
```diff
@@ -191,98 +185,110 @@ func (s *cpFileSuite) TestGet(c *C) {
...
-	})
+	}
+	assert.Equal(t, expect, cp)
 
-	cp, err = s.cpdb.Get(ctx, "`db3`.`not-exists`")
-	c.Assert(cp, IsNil)
-	c.Assert(errors.IsNotFound(err), IsTrue)
+	cp, err = cpdb.Get(ctx, "`db3`.`not-exists`")
+	require.Nil(t, cp)
+	assert.True(t, errors.IsNotFound(err))
```
Comment (tisonkun) [inline br/pkg/lightning/checkpoints/checkpoints_file_test.go:206]:
```suggestion
	require.True(t, errors.IsNotFound(err))
```

We can consistently use `require` for fail fast. If the codebase exists two variants while no explicit benefits, it increases understanding overhead.

ditto others

Comment (disksing) [reply br/pkg/lightning/checkpoints/checkpoints_file_test.go:206]: ok.

**Review by tisonkun — APPROVED** [review]
LGTM.

/cc @hawkingrei

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**